iothub/device/src/Transport/AmqpIoT/AmqpIoTSession.cs

```diff
@@ -243,7 +269,7 @@ TimeSpan timeout
 
             AmqpLinkSettings amqpLinkSettings = new AmqpLinkSettings
             {
-                LinkName = CommonResources.GetNewStringGuid(linkSuffix),
+                LinkName = linkSuffix,
```

Comment:
These don't need to be unique?

Reply:
It is unique as above.

Reply:
See above : I don't see how this is unique.

Reply:
This is unique because of the combination of device id and link type. As device ID is unique, this will stay unique.

Reply:
@prmathur-microsoft A device can reconnect with a different session on the same connection. This means that for any reuse of a connection we will have duplicate names in different session contexts.